### DISCUSSION TITLE
Can someone walk me through configuring this guy?

### DISCUSSION THREAD

**Discussion by dreinert1** [crissNb/Dynamic-Island-Sketchybar#28] [General]
Macbook Pro 2023 14" M2 Max. 

The idle notch was too big and I wanted to configure it to be a bit smaller. I'm extremely new to github stuff and until like a month ago I hadn't ever touched the terminal. In addition to wanting to adjust the size, I figured I will eventually want to enable/disable other features, and so I'm asking here.

Is there a simple way to configure? Maybe a UI with input fields or sliders? If that doesn't exist, could someone ELI5 how to configure this thing?

Thanks in advance. I appreciate you guys and respect what you do because it's not for laymen.

Best,
Dan

**Comment by crissNb** [comment]
Hello,

All the configurations can be done by adjusting `userconfig.sh` (initially when you're installing the program, you have made a copy of `userconfig.sh` to `~/.config/sketchybar/`, so you should be able to locate this file under `~/.config/sketchybar/userconfig.sh`).

Unfortunately, there isn't any UI for configuration (and I currently don't have any plans on implementing UI for configurations). I hate to say it, but the config is quite messy as well and you'll probably have to experiment with various values a lot just to get it right.

Primarily when you're adjusting the size of the notch, you want to take a look at the following variables in `userconfig.sh`:
```bash
P_DYNAMIC_ISLAND_MAX_HEIGHT=90
P_DYNAMIC_ISLAND_DEFAULT_HEIGHT=65
P_DYNAMIC_ISLAND_DEFAULT_WIDTH=205
```
These values determine the "base size" of the notch.

As of right now, there's quite a hacky solution in getting the various island sizes. What I mean is that when you're adjusting the notch size, you should probably also adjust values like:
```bash
P_DYNAMIC_ISLAND_APPSWITCH_EXPAND_WIDTH=150
...
P_DYNAMIC_ISLAND_VOLUME_EXPAND_WIDTH=192
```

and so on... You can find the correct values by trial and error. Once you found the "right value", you'll see a smooth animation (as I said, it's quite a hacky solution right now, oops.)

> I figured I will eventually want to enable/disable other features

This can be done by enabling or disabling islands in `userconfig.sh`. `1` for enable, and `0` for disable.

Once you make a change to your config, restart sketchybar service to take effect:

```bash
brew services restart sketchybar
```

In case you do get a satisfactory config in the end, please let me know so that we can add it to the repository for the other users.
